ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FSharpIdentifierTooltipProvider.fs

```diff
@@ -23,12 +23,12 @@ open JetBrains.Util
 let [<Literal>] RiderTooltipSeparator = "_RIDER_HORIZONTAL_LINE_TOOLTIP_SEPARATOR_"
 
 [<SolutionComponent>]
-type FSharpIdentifierTooltipProvider(lifetime, solution, presenter, xmlDocService: FSharpXmlDocService) =
+type FSharpIdentifierTooltipProvider(lifetime, solution, presenter, xmlDocService: FSharpXmlDocService, reactorMonitor) =
```

Comment:
I don't think I'm in favour of passing the reactor monitor from the features code like this since it'll make us have to pass it like this in every other component doing similar things. I'd prefer it being hidden in the checker service just to make it easier to write new code in future.

Reply:
This code isn't using the checker service - it's calling `GetStructuredToolTipText` on `FSharpCheckFileResults`. Would you accept using `IFcsReactorMonitor` in this instance?

Reply:
Could we have the reactor monitor easily accessible via the checker service, so it'd be something like `token.CheckerService.ReactorMonitor` inside `GetFSharpToolTipText`?